go/vt/vttablet/tabletserver/schema/engine.go

```diff
@@ -151,8 +151,36 @@ func (se *Engine) syncVTDatabase(ctx context.Context, conn *dbconnpool.DBConnect
 		}
 		return conn.ExecuteFetch(query, maxRows, wantFields)
 	}
+	var _ sidecardb.ReSetSuperReadOnlyHook = func(ctx context.Context) (err error) {
+		log.Infof("resetting hook for super read only...")
+		if _, err := dbaConn.ExecuteFetch("SET GLOBAL super_read_only='ON'", 1, false); err != nil {
+			log.Infof("Not able to set super_read_only user... This can cause errant GTIDs in future.")
+			return err
+		}
+		return nil
+	}
+	var _ sidecardb.SetSuperReadOnlyHook = func(ctx context.Context) (needsReset bool, err error) {
+		if !dbaConn.IsMariaDB() {
+			if err := dbaConn.WriteComQuery("SELECT @@global.super_read_only"); err != nil {
+				log.Infof("Not able to select super_read_only.")
+				return false, err
+			}
+			res, _, _, err := dbaConn.ReadQueryResult(1, false)
+			if err == nil && len(res.Rows) == 1 {
+				sro := res.Rows[0][0].ToString()
+				if sro == "1" || sro == "ON" {
+					log.Infof("setting super read only to false...")
+					if _, err = dbaConn.ExecuteFetch("SET GLOBAL read_only='OFF'", 1, false); err != nil {
+						return false, err
+					}
+					return true, nil
+				}
+			}
+		}
```

Comment:
It feels like we have many instances of this same code. We should find a way to leverage functions that we can re-use.